src/main/java/com/onfleet/demo/homework/collection/Tasklist.java

```diff
@@ -43,14 +70,110 @@
    * @param driver Driver object for this task-list.
    * @param tasks Collection of tasks for this driver.
    */
-  Tasklist(final @NotNull Driver driver,
-           final @NotNull Collection<Task> tasks) {
+  public Tasklist(final @NotNull Driver driver,
+                  final @NotNull Collection<Task> tasks) {
     this.driver = driver;
 
     // note: copied here to avoid mutation of underlying private value -
     // chose not to use `unmodifiableCollection` because we modify tasks as we go
     this.assignedTasks = new ArrayList<>(tasks.size());
     this.assignedTasks.addAll(tasks);
+    this.taskCount = this.assignedTasks.size();
+
+    Task lastTask = null;
+    double currentKnownDistance = this.knownDistance;
+    for (final Task task : tasks) {
+      currentKnownDistance += calculateDistanceForPoints(lastTask, task);
+      lastTask = task;
+    }
+    this.lastAssignedTask = lastTask;
+    this.knownDistance = currentKnownDistance;
+    recalculateLoadEstimate();
+  }
+
+  // -- static API -- //
+  /**
+   * Calculate the distance cost between two tasks using their
+   * underlying geopoints.
+   *
+   * @param start Start geopoint.
+   * @param finish End geopoint.
+   * @return The value to use for the distance.
+   */
+  public static double calculateDistanceForPoints(final Task start, final Task finish) {
```

Comment:
the last curly is the close of the function. it just looks awkward because i'm capturing `difference` so i can `assert difference >= 0.0`, which will fail at runtime if the condition isn't met.

if i didn't need to do that i'd `return [...]` directly.